chia/wallet/vc_wallet/cr_cat_wallet.py

```diff
@@ -0,0 +1,927 @@
+                sent=uint32(0),
+                spend_bundle=claim_bundle,
+                additions=claim_bundle.additions(),
+                removals=claim_bundle.removals(),
+                wallet_id=self.id(),
+                sent_to=[],
+                trade_id=None,
+                type=uint32(TransactionType.OUTGOING_TX.value),
+                name=claim_bundle.name(),
+                memos=list(compute_memos(claim_bundle).items()),
+            ),
+            TransactionRecord(
+                confirmed_at_height=uint32(0),
+                created_at_time=uint64(int(time.time())),
+                to_puzzle_hash=await self.wallet_state_manager.main_wallet.get_puzzle_hash(False),
+                amount=uint64(sum(c.amount for c in coins)),
+                fee_amount=fee,
+                confirmed=False,
+                sent=uint32(0),
+                spend_bundle=None,
+                additions=[],
+                removals=[],
+                wallet_id=self.id(),
+                sent_to=[],
+                trade_id=None,
+                type=uint32(TransactionType.INCOMING_TX.value),
+                name=claim_bundle.name(),
+                memos=list(compute_memos(claim_bundle).items()),
+            ),
+            *(dataclasses.replace(tx, spend_bundle=None) for tx in vc_txs),
+            *((dataclasses.replace(chia_tx, spend_bundle=None),) if chia_tx is not None else []),
+        ]
+
+    async def match_puzzle_info(self, puzzle_driver: PuzzleInfo) -> bool:
+        if (
+            AssetType(puzzle_driver.type()) == AssetType.CAT
+            and puzzle_driver["tail"] == self.info.limitations_program_hash
+        ):
+            inner_puzzle_driver: Optional[PuzzleInfo] = puzzle_driver.also()
+            if inner_puzzle_driver is None:
+                raise ValueError("Malformed puzzle driver passed to CRCATWallet.match_puzzle_info")  # pragma: no cover
+            return (
+                AssetType(inner_puzzle_driver.type()) == AssetType.CR
+                and [bytes32(provider) for provider in inner_puzzle_driver["authorized_providers"]]
+                == self.info.authorized_providers
+                and ProofsChecker.from_program(uncurry_puzzle(inner_puzzle_driver["proofs_checker"]))
+                == self.info.proofs_checker
+            )
+        return False
+
+    async def get_puzzle_info(self, asset_id: bytes32) -> PuzzleInfo:
+        return PuzzleInfo(
+            {
+                "type": AssetType.CAT.value,
+                "tail": "0x" + self.info.limitations_program_hash.hex(),
+                "also": {
+                    "type": AssetType.CR.value,
+                    "authorized_providers": ["0x" + provider.hex() for provider in self.info.authorized_providers],
+                    "proofs_checker": self.info.proofs_checker.as_program(),
+                },
+            }
+        )
+
+    async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
```

Comment:
Might be good to note which puzzles this matches against

Reply:
Added a comment